### PR TITLE
PLAT-110254: InputField: Fix cursor jumps unexpectedly when the cursor is at the end of the text and clicking the field

### PR DESCRIPTION
**Pull request by YunhoLee0**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
The input tag within the InputField is double focused by the onMouseDown handler and clicking it.
So when the cursor is outside the field and focused by the onMouseDown handler, the text inside the field moves so that the cursor shows. And then, the mouse click is actually applied, the cursor seems to be jumped unexpectedly.
It needs to be fixed.

### Resolution
Prevent scroll while focusing when the mouse button is downed.

### Additional Considerations
It might generate some side-effects, so should be tested with the screenshot test and existed TCs.

### Links
PLAT-110254

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 - `sandstone/FormCheckboxItem` to show correct color for `slotBefore` icon in disabled state when focused
 - `sandstone/ImageItem` to resize the image properly
 - `sandstone/Input` button label when default value is `0`
+- `sandstone/InputField` cursor behavior to not jump unexpectedly
```
Comment (MikyungKim) [inline CHANGELOG.md:28]:
```suggestion
- `sandstone/InputField` cursor to not jump unexpectedly when mouse down
```

File: Input/InputFieldSpotlightDecorator.js [hunk 2]
```diff
@@ -144,7 +144,8 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
 
 			this.state = {
 				focused: null,
-				node: null
+				node: null,
+				mouseDowned: null
```
Comment (MikyungKim) [inline Input/InputFieldSpotlightDecorator.js:148]:
```suggestion
				fromMouse: false
```

File: Input/InputFieldSpotlightDecorator.js [hunk 3]
```diff
@@ -180,7 +181,8 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
 				Spotlight.getCurrent() !== this.state.node &&
 				(this.paused.isPaused() || !Spotlight.isPaused())
 			) {
-				this.state.node.focus();
+				if (this.state.mouseDowned) this.state.node.focus({preventScroll: true});
```
Comment (MikyungKim) [inline Input/InputFieldSpotlightDecorator.js:184]: Please add {} to if and else.

File: Input/InputFieldSpotlightDecorator.js [hunk 4]
```diff
@@ -201,8 +203,14 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
 			}
 		};
 
-		focus = (focused, node) => {
-			this.setState({focused, node});
+		focus = (focused, node, calledFromMouseDown) => {
```
Comment (MikyungKim) [inline Input/InputFieldSpotlightDecorator.js:206]:
```suggestion
		focus = (focused, node, fromMouse) => {
```

File: CHANGELOG.md (hunk 1 above)
Comment (YunhoLee0) [reply CHANGELOG.md]: Thank you for correcting.

File: Input/InputFieldSpotlightDecorator.js (hunk 2 above)
Comment (YunhoLee0) [reply Input/InputFieldSpotlightDecorator.js]: 🆗

File: Input/InputFieldSpotlightDecorator.js (hunk 3 above)
Comment (YunhoLee0) [reply Input/InputFieldSpotlightDecorator.js]: I've fixed it with the brace.

File: Input/InputFieldSpotlightDecorator.js (hunk 4 above)
Comment (YunhoLee0) [reply Input/InputFieldSpotlightDecorator.js]: 🆗

File: CHANGELOG.md [hunk 5]
```diff
@@ -28,6 +28,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 - `sandstone/FormCheckboxItem` to show correct color for `slotBefore` icon in disabled state when focused
 - `sandstone/ImageItem` to resize the image properly
 - `sandstone/Input` button label when default value is `0`
+- `sandstone/InputField` cursor not to jump unexpectedly when mouse down
```
Comment (MikyungKim) [inline CHANGELOG.md:31]: Please rebase your branch with the latest develop and put the log in the right place.

File: Input/InputFieldSpotlightDecorator.js [hunk 6]
```diff
@@ -201,8 +206,14 @@ const InputSpotlightDecorator = hoc(defaultConfig, (config, Wrapped) => {
...
 		};
 
-		focus = (focused, node) => {
-			this.setState({focused, node});
+		focus = (focused, node, fromMouse) => {
+			this.setState(() => (
+				{
+					focused,
+					node,
+					fromMouse: fromMouse
```
Comment (MikyungKim) [inline Input/InputFieldSpotlightDecorator.js:214]:
```suggestion
					fromMouse
```

Comment (MikyungKim) [reply Input/InputFieldSpotlightDecorator.js:214]: I think we can just add `fromMouse` next to the `node`.

File: CHANGELOG.md (hunk 5 above)
Comment (YunhoLee0) [reply CHANGELOG.md]: OK. it's rebased and fixed.

File: Input/InputFieldSpotlightDecorator.js (hunk 6 above)
Comment (YunhoLee0) [reply Input/InputFieldSpotlightDecorator.js]: Thank you. And I changed it to not use the function like the previous one.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**